tests/unittest_builder.py

```diff
@@ -790,5 +794,62 @@ def test_parse_module_with_invalid_type_comments_does_not_crash():
     assert isinstance(node, nodes.Module)
 
 
+class HermeticInterpreterTest(unittest.TestCase):
+    """Modeled on https://github.com/PyCQA/astroid/pull/1207#issuecomment-951455588"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Simulate a hermetic interpreter environment having no code on the filesystem."""
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            sys.path.append(tmp_dir)
+
+            # Write a python file and compile it to .pyc
+            cls.code_snippet = "def func():  return 42"
```

Comment:
To make this test have even more value, we would need to come up with some code that gets inferred differently when we get its "partial representation". This code is too simple for that. But we can't use builtins either, because we would have to delete builtins from the filesystem. This is why the author speculated about writing a "littletest" engineered to have inference ambiguity. But even if we engineered that, the _difference_ might evaporate over time as inference changes.

Reply:
```suggestion
            # To make this test have even more value, we would need to come up with some 
            # code that gets inferred differently when we get its "partial representation".
            # This code is too simple for that. But we can't use builtins either, because we would 
            # have to delete builtins from the filesystem.  But even if we engineered that, 
            # the difference might evaporate over time as inference changes.
            cls.code_snippet = "def func():  return 42"
```